File: Lab5.Web/src/infrastructure/repositories/InMemoryRentalRepository.js
export class InMemoryRentalRepository {
  constructor() {
    this.rentals = [];
  }

  getActiveRentalsCountForUser(userId) {
    return this.rentals.filter((r) => r.userId === userId).length;
  }

  getActiveRentalsCountForBook(bookId) {
    return this.rentals.filter((r) => r.bookId === bookId).length;
  }

  add(rental) {
    this.rentals.push(rental);
  }
}

File: Lab5/Infrastructure/Repositories/InMemoryRentalRepository.cs
using Lab5.Domain.Entities;
using Lab5.Domain.Interfaces;

namespace Lab5.Infrastructure.Repositories;

public sealed class InMemoryRentalRepository : IRentalRepository
{
    private readonly List<RentalRecord> _rentals = [];

    public int GetActiveRentalsCountForUser(Guid userId) => _rentals.Count(x => x.UserId == userId);

    public int GetActiveRentalsCountForBook(Guid bookId) => _rentals.Count(x => x.BookId == bookId);

    public void Add(RentalRecord rentalRecord) => _rentals.Add(rentalRecord);
}
